tests/Util/AutoloaderUtilTest.php

```diff
@@ -31,39 +30,39 @@ public function testGetPathForFutureClass()
         $composerJson = [
             'autoload' => [
                 'psr-4' => [
-                    'Also\\In\\Src\\' => 'src/SubDir',
-                    'App\\' => 'src/',
-                    'Other\\Namespace\\' => 'lib',
-                    '' => 'fallback_dir',
+                    'Also\\In\\Src\\' => '/src/SubDir',
+                    'App\\' => '/src',
+                    'Other\\Namespace\\' => '/lib',
+                    '' => '/fallback_dir',
                 ],
                 'psr-0' => [
-                    'Psr0\\Package' => 'lib/other',
+                    'Psr0\\Package' => '/lib/other',
                 ],
             ],
         ];
 
-        $fs = new Filesystem();
-        if (!file_exists(self::$currentRootDir)) {
-            $fs->mkdir(self::$currentRootDir);
-        }
+        $reflection = new \ReflectionClass(AutoloaderUtil::class);
+        $method = $reflection->getMethod('getClassLoader');
+        $method->setAccessible(true);
 
-        $fs->remove(self::$currentRootDir.'/vendor');
-        file_put_contents(
-            self::$currentRootDir.'/composer.json',
-            json_encode($composerJson, JSON_PRETTY_PRINT)
-        );
-        $process = new Process('composer dump-autoload', self::$currentRootDir);
-        $process->run();
-        if (!$process->isSuccessful()) {
-            throw new \Exception('Error running composer dump-autoload: '.$process->getErrorOutput());
-        }
+        $autoloaderUtil = new AutoloaderUtil();
 
+        $autoloader = $method->invoke($autoloaderUtil);
+        foreach ($composerJson['autoload'] as $psr => $dirs) {
+            foreach ($dirs as $prefix => $path) {
+                if ($psr == 'psr-4') {
+                    $autoloader->addPsr4($prefix, self::$currentRootDir.$path);
+                } else {
+                    $autoloader->add($prefix, self::$currentRootDir.$path);
+                }
+            }
+        }
+        $autoloader->setUseIncludePath(true);
```

Comment:
mutating the actual autoloader of PHP is a bad idea, as it will impact the test process (until the end of the testsuite).

a better idea would be to create a `Composer\ClassLoader` instance with your fake configuration, and put it in the `$autoloaderUtil` private property (as if the class loader had already been located previously).
And then, you could have a separate test triggering the actual class loader retrieval code, but without altering the instance itself.